src/num_methods/simulated_annealing.md

```diff
@@ -0,0 +1,198 @@
+
+Simulated annealing, literally simulates this process. We start off with some random state (material) and set a high temperature (heat it up). Now, the algorithm is ready to accept states which have a higher energy than the current state, as it is motivated by the high value of $T$. This prevents the algorithm from getting stuck inside local minimas and move towards the global minima. As time progresses, the algorithm cools down and refuses the states with higher energy and moves into the closest minima it has found.
+
+
+<center>
+<img src="https://upload.wikimedia.org/wikipedia/commons/d/d5/Hill_Climbing_with_Simulated_Annealing.gif" width="800px">
+<br>
+<i>A visual representation of simulated annealing, searching for the maxima of this function with multiple local maxima.</i>.
+<br>
+<i>This <a href="https://upload.wikimedia.org/wikipedia/commons/d/d5/Hill_Climbing_with_Simulated_Annealing.gif">gif</a> by [Kingpin13](https://commons.wikimedia.org/wiki/User:Kingpin13) is distributed under <a href="https://creativecommons.org/publicdomain/zero/1.0/deed.en">CC0 1.0</a></i> license.
+</center>
+
+## Probability Acceptance Function(PAF)
+
+$P(E,E_{next},T) = 
+    \begin{cases}
+       \text{True} &\quad\text{if }  \exp{\frac{-(E_{next}-E)}{T}} \ge random(0,1) \\
+       \text{False} &\quad\text{otherwise}\\
+     \end{cases}$
+
+This function takes in the current state, the next state and the Temperature , returning a boolean value, which tells our search whether it should move to $s_{next}$ or stay at $s$. Note that for $E_{next} < E$ , this function will always return True. 
+
+```cpp
+bool P(double E,double E_next,double T){
+    double prob = (double)rand()/RAND_MAX; // Generate a random number between 0 and 1
+    if(exp(-(E_next-E)/T) > prob) return true;
+    else return false;
+}
+```
+## Code Template
+
+```cpp
+class state{
+    public:
+    state(){
+        // Generate the initial state
+    }
+    state(state& s){
+        // Implement the copy constructor
+    }
+    state next(){
+        state s_next = state(*this);
+
+        // Modify s_next to the neighbouring state
+
+        return s_next;
+    }
+    double E(){
+        // Implement the cost function here
+    };
+};
+
+pair<double,state> simAnneal(){
+    state s = state();
+    state best = state(s);
+    double T = 1000; // Initial temperature
+    double u = 0.99; // decay rate
```

Comment:
It seems to be the first time the "decay" is introduced. It should be mentioned earlier, as well as some explanation to why the rate of decay is exponential.